src/plugins/home/server/services/sample_data/sample_data_installer.ts

```diff
@@ -135,18 +135,45 @@ export class SampleDataInstaller {
     } catch (err) {
       // ignore delete errors
     }
+
+    try {
+      await this.esClient.asCurrentUser.indices.deleteIndexTemplate({
+        name: index,
+      });
+    } catch (err) {
+      // ignore error
+    }
   }
 
   private async installDataIndex(dataset: SampleDatasetSchema, dataIndex: DataIndexSchema) {
     const index = createIndexName(dataset.id, dataIndex.id);
     try {
-      await this.esClient.asCurrentUser.indices.create({
-        index,
-        body: {
-          settings: { index: { number_of_shards: 1, auto_expand_replicas: '0-1' } },
-          mappings: { properties: dataIndex.fields },
-        },
-      });
+      if (dataIndex.isDataStream) {
+        const request = {
+          name: index,
+          body: {
+            template: {
+              settings: { number_of_shards: 1, auto_expand_replicas: '0-1' },
+              mappings: { properties: dataIndex.fields },
+            },
+            index_patterns: [index],
+            data_stream: {},
+          },
+        };
+        await this.esClient.asCurrentUser.indices.putIndexTemplate(request);
+
+        await this.esClient.asCurrentUser.indices.createDataStream({
+          name: index,
+        });
+      } else {
+        await this.esClient.asCurrentUser.indices.create({
+          index,
+          body: {
+            settings: { index: { number_of_shards: 1, auto_expand_replicas: '0-1' } },
+            mappings: { properties: dataIndex.fields },
+          },
```

Comment:
nit: maybe this could be one thing `templateBodyConfig`.

```suggestion
            template: templateBodyConfig,
            index_patterns: [index],
            data_stream: {},
          },
        };
        await this.esClient.asCurrentUser.indices.putIndexTemplate(request);

        await this.esClient.asCurrentUser.indices.createDataStream({
          name: index,
        });
      } else {
        await this.esClient.asCurrentUser.indices.create({
          index,
          body: templateBodyConfig,
```